covid19-etl/etl/idph_regional_icu_capacity.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import os
+import re
+from contextlib import closing
+
+from etl import base
+from utils.format_helper import (
+    derived_submitter_id,
+    format_submitter_id,
+)
+from utils.metadata_helper import MetadataHelper
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+def strip_prefix(region):
+    # removes any digits, spaces and dashes from begining of string
+    # used to cleanup the hospital region
+    return re.sub(r"^[\d\s-]+", "", region)
+
+
+class IDPH_REGIONAL_ICU_CAPACITY(base.BaseETL):
+    def __init__(self, base_url, access_token, s3_bucket):
+        super().__init__(base_url, access_token, s3_bucket)
+
+        self.program_name = "open"
+        self.project_code = "IDPH-Regional_ICU_Capacity"
+        self.metadata_helper = MetadataHelper(
+            base_url=self.base_url,
+            program_name=self.program_name,
+            project_code=self.project_code,
+            access_token=access_token,
+        )
+
+        self.country = "US"
+        self.state = "IL"
+        self.etlJobDate = datetime.date.today().strftime("%Y-%m-%d")
+        self.summary_locations = []
+        self.summary_clinicals = []
+
+    def files_to_submissions(self):
+        """
+        Reads JSON file and convert the data to Sheepdog records
+        """
+
+        today = datetime.date.today()
+        latest_submitted_date = self.metadata_helper.get_latest_submitted_date_idph()
+        if latest_submitted_date == today:
+            print("Nothing to submit: today and latest submitted date are the same.")
+            return
+        today_str = today.strftime("%Y%m%d")
+
+        print(f"Getting data for date: {today_str}")
+        url = "https://idph.illinois.gov/DPHPublicInformation/api/COVIDExport/GetHospitalizationResultsRegion"
+        self.parse_file(latest_submitted_date, url)
+
+    def parse_file(self, latest_submitted_date, url):
+        """
+        Converts a JSON files to data we can submit via Sheepdog. Stores the
+        records to submit in `self.summary_locations` and `self.summary_clinicals`.
+
+        Args:
+            latest_submitted_date (date): the date of latest available "summary_clinical" for project
+            url (str): URL at which the JSON file is available
+        """
+        print("Getting data from {}".format(url))
+        with closing(self.get(url, stream=True)) as r:
+            data = r.json()
+            date = self.etlJobDate
+
+            for region in data:
+                (summary_location, summary_clinical) = self.parse_region(date, region)
+                self.summary_locations.append(summary_location)
```

Comment:
i'm guessing you only need to submit the summary_locations once. Maybe you could get the list of existing summary_locations from guppy and avoid re-submitting the ones that already exist

Reply:
This is currently blocked by Tube issues; will do in later PR